### PR TITLE
Add swiftsourceinfo file to worker incremental outputs

### PR DESCRIPTION
**Pull request by BalestraPatrick**
After bumping our `rules_swift` version to d69bc4080d03bac10339ab3f8c105c3cdc7c3adf, we noticed some failures possibly related to the changes in 9a1073d28b6c3583428f05aaaa2091833320e71d.

About 10-20% of our builds started failing on CI and locally with errors such as:
```
ERROR: /path/to/repo/BUILD.bazel:9:14: output 'container/Wiring/ContainerWiring.swiftsourceinfo' was not created
```

Digging into the execution log, it seems like this file was in `listed_outputs` but not present in `actual_outputs` in some cases. We always build with sandbox on.

Basically in https://github.com/bazelbuild/rules_swift/pull/523/files we wanted to add support for `swiftsourceinfo`. Even though that PR wasn’t merged as far as I can see, the cherry-picked commit from upstream (https://github.com/bazelbuild/rules_swift/pull/692) didn’t even contain the change to `tools/worker/output_file_map.cc`. The following patch fixes it for me on a local machine that previously was failing 100% of the builds with errors like the one above.

### CODE REVIEW

**Comment by acecilia** [comment]
I am experiencing the same problem when building a mac CLI with bazel (using `swift_library` rule). Tried using the latest commit on master, but did not help

**Comment by keith** [comment]
huh, can you file an issue with a repro case?

**Comment by acecilia** [comment]
Sorry for late reply. I was now trying to reproduce the problem and provide a reproducible case, but seems I cant anymore 😕 Maybe I was doing something wrong, or could be that during last month other changes in my project fixed the problem
